src/tree/containerizedFunctionApp/ResolvedContainerizedFunctionAppResourceBase.ts

```diff
@@ -0,0 +1,150 @@
+        const appSettings: StringDictionary = await client.listApplicationSettings();
+
+        return [runFromPackageKey, 'WEBSITE_RUN_FROM_ZIP'].some(key => appSettings.properties && envUtils.isEnvironmentVariableSet(appSettings.properties[key]));
+    }
+
+    public async getHostRequest(): Promise<FuncHostRequest> {
+        return { url: `https://${nonNullValueAndProp(this.site, 'defaultHostName')}` }
+    }
+
+    public async getHostJson(context: IActionContext): Promise<IParsedHostJson> {
+        let result: IParsedHostJson | undefined = this._cachedHostJson;
+        if (!result) {
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            let data: any;
+            const version: FuncVersion = await this.getVersion(context);
+            result = parseHostJson(data, version);
+            this._cachedHostJson = result;
+        }
+        return result;
+    }
+
+    public async getVersion(context: IActionContext): Promise<FuncVersion> {
+        let result: FuncVersion | undefined = this._cachedVersion;
+        if (result === undefined) {
+            let version: FuncVersion | undefined;
+            try {
+                const proxyTree: ContainerTreeItem = this as unknown as ContainerTreeItem;
+                const client = await (new ContainerAppSettingsClientProvider(proxyTree, proxyTree.subscription).createClient(context));
+                const appSettings: StringDictionary = await client.listApplicationSettings();
+                version = tryParseFuncVersion(appSettings.properties && appSettings.properties.FUNCTIONS_EXTENSION_VERSION);
+            } catch {
+                // ignore and use default
+            }
+            result = version || latestGAVersion;
+            this._cachedVersion = result;
+        }
+        return result
+    }
+
+    public async getApplicationSettings(context: IActionContext): Promise<ApplicationSettings> {
+        const proxyTree: ContainerTreeItem = this as unknown as ContainerTreeItem;
+        const client = await (new ContainerAppSettingsClientProvider(proxyTree, proxyTree.subscription).createClient(context));
+        const appSettings: StringDictionary = await client.listApplicationSettings();
+        return appSettings.properties || {};
+    }
+
+    public async setApplicationSetting(context: IActionContext, key: string, value: string): Promise<void> {
+        const proxyTree: ContainerTreeItem = this as unknown as ContainerTreeItem;
+        const client = await (new ContainerAppSettingsClientProvider(proxyTree, proxyTree.subscription).createClient(context));
+        const appSettings: StringDictionary = await client.listApplicationSettings();
+        if (!appSettings.properties) {
+            appSettings.properties = {};
+        }
+        appSettings.properties[key] = value;
+        await client.updateApplicationSettings(appSettings);
+    }
+}
+
```

Comment:
You'll need to implement `pickTreeItemImpl` for when people do things in the command palette.